src/main/java/subway/service/SectionService.java

```diff
@@ -0,0 +1,73 @@
+package subway.service;
+
+import java.util.List;
+import java.util.Optional;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+import subway.entity.Section;
+import subway.entity.group.SectionGroup;
+import subway.repository.SectionRepository;
+
+@Service
+@Transactional
+public class SectionService {
+
+    private final SectionRepository sectionRepository;
+
+    public SectionService(SectionRepository sectionRepository) {
+        this.sectionRepository = sectionRepository;
+    }
+
+    public Section create(long lineId, long upStationId, long downStationId,
+        int distance) {
+
+        return save(lineId, upStationId, downStationId, distance);
+    }
+
+    private Section save(long lineId, long upStationId, long downStationId,
+        int distance) {
+        return sectionRepository.save(
+            new Section(
+                lineId,
+                upStationId,
+                downStationId,
+                distance
+            )
+        );
+    }
+
+    public Section add(long lineId, long upStationId, long downStationId,
+        int distance) {
+
+        validAddSection(lineId, upStationId, downStationId);
+
+        return save(lineId, upStationId, downStationId, distance);
+    }
+
+    private void validAddSection(long lineId, long upStationId, long downStationId) {
+
+        SectionGroup group = SectionGroup.of(findAllByLineId(lineId));
+
+        group.validateAdd(upStationId, downStationId);
+    }
+
+    public List<Section> findAllByLineId(Long lineId) {
+        Optional<List<Section>> list = sectionRepository.findAllByStationLineId(
+            lineId);
+
+        if (list.isEmpty()) {
+            throw new IllegalArgumentException(lineId + " id 값을 가지는 노선 구간이 존재하지 않습니다.");
+        }
```

Comment:
```suggestion
        List<Section> list = sectionRepository.findAllByStationLineId(lineId)
            .orElseThrow(() -> throw new IllegalArgumentException(lineId + " id 값을 가지는 노선 구간이 존재하지 않습니다."));
```

이렇게 구현할 수 있어요 😄 
[참고](http://homoefficio.github.io/2019/10/03/Java-Optional-%EB%B0%94%EB%A5%B4%EA%B2%8C-%EC%93%B0%EA%B8%B0/) 도 읽어보시면 좋을 것 같습니다 👍